library/systemd/src/modules/systemd_service.rb

```diff
@@ -62,32 +65,68 @@ module Yast
   #
   #    service = Yast::SystemdService.find('sshd', :type=>'Type')
   #    service.properties.type  # 'simple'
-  #
-  ##
-
-  class SystemdServiceNotFound < StandardError
-    def initialize(service_name)
-      super "Service unit '#{service_name}' not found"
-    end
-  end
-
   class SystemdServiceClass < Module
+    Yast.import "Stage"
+    include Yast::Logger
+
     UNIT_SUFFIX = ".service".freeze
 
-    def find(service_name, properties = {})
+    # @param service_name [String] "foo" or "foo.service"
+    # @param propmap [SystemdUnit::PropMap]
+    # @return [Service,nil] `nil` if not found
+    def find(service_name, propmap = {})
       service_name += UNIT_SUFFIX unless service_name.end_with?(UNIT_SUFFIX)
-      service = Service.new(service_name, properties)
+      service = Service.new(service_name, propmap)
       return nil if service.properties.not_found?
       service
     end
 
-    def find!(service_name, properties = {})
-      find(service_name, properties) || raise(SystemdServiceNotFound, service_name)
+    # @param service_name [String] "foo" or "foo.service"
+    # @param propmap [SystemdUnit::PropMap]
+    # @return [Service]
+    # @raise [SystemdServiceNotFound]
+    def find!(service_name, propmap = {})
+      find(service_name, propmap) || raise(SystemdServiceNotFound, service_name)
+    end
+
+    # @param service_names [Array<String>] "foo" or "foo.service"
+    # @param propmap [SystemdUnit::PropMap]
+    # @return [Array<Service,nil>] `nil` if not found, [] if no can do
+    # @raise [SystemdServiceNotFound] if an unexpected problem occurs
+    def find_many_at_once(service_names, propmap = {})
```

Comment:
btw does it make sense to have it public? or it is just private speed up method?